cdap-common/src/test/java/co/cask/cdap/common/zookeeper/coordination/ResourceCoordinatorTest.java

```diff
@@ -236,4 +245,22 @@ public InetSocketAddress getSocketAddress() {
       }
     };
   }
+
+  private Collection<PartitionReplica> waitForChange(BlockingQueue<Collection<PartitionReplica>> assignmentQueue,
```

Comment:
Rather than just extending the timeout, we add multiple possible pollings, based on timeout, during larger period of time, 30s for now.
So, in best case in will return less than 5 sec but could wait as long as 30s before an element occur.